### PR TITLE
EPMRPP-50431 || Google analytics client impl

### PR DESCRIPTION
**Pull request by Yumfriez**

### CODE REVIEW

File: src/test/java/com/epam/reportportal/service/analytics/GoogleAnalyticsTest.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+/**
+ * @author <a href="mailto:ivan_budayeu@epam.com">Ivan Budayeu</a>
+ */
+public class GoogleAnalyticsTest {
+
+	private final Scheduler scheduler = Schedulers.from(Executors.newSingleThreadExecutor());
+	private final HttpClient httpClient = mock(HttpClient.class);
+
+	@Test
+	void send() throws IOException {
```
Comment (HardNorth) [inline src/test/java/com/epam/reportportal/service/analytics/GoogleAnalyticsTest.java:44]: Please specify comprehensive, descriptive test name.

File: src/test/java/com/epam/reportportal/service/analytics/GoogleAnalyticsTest.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+import java.io.IOException;
+import java.util.concurrent.Executors;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.*;
+
+/**
+ * @author <a href="mailto:ivan_budayeu@epam.com">Ivan Budayeu</a>
+ */
+public class GoogleAnalyticsTest {
```
Comment (HardNorth) [inline src/test/java/com/epam/reportportal/service/analytics/GoogleAnalyticsTest.java:38]: Please also add connection error test. We should not affect launch/tests in this case.

Comment (Yumfriez) [reply src/test/java/com/epam/reportportal/service/analytics/GoogleAnalyticsTest.java:39]: added

File: src/test/java/com/epam/reportportal/service/analytics/GoogleAnalyticsTest.java (hunk 1 above)
Comment (Yumfriez) [reply src/test/java/com/epam/reportportal/service/analytics/GoogleAnalyticsTest.java]: renamed

File: src/main/java/com/epam/reportportal/utils/properties/DefaultProperties.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.epam.reportportal.utils.properties;
+
+/**
```
[the remark this answers is not in the document]
Comment (Yumfriez) [reply src/main/java/com/epam/reportportal/utils/properties/DefaultProperties.java:19]: added

File: src/main/java/com/epam/reportportal/utils/properties/ClientProperties.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.epam.reportportal.utils.properties;
+
+/**
```
[the remark this answers is not in the document]
Comment (Yumfriez) [reply src/main/java/com/epam/reportportal/utils/properties/ClientProperties.java:19]: added

File: src/main/java/com/epam/reportportal/service/analytics/item/AnalyticsItem.java [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.epam.reportportal.service.analytics.item;
+
+import java.util.Map;
+
+/**
```
[the remark this answers is not in the document]
Comment (Yumfriez) [reply src/main/java/com/epam/reportportal/service/analytics/item/AnalyticsItem.java:21]: added

File: src/main/java/com/epam/reportportal/service/analytics/item/AnalyticsEvent.java [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+ */
+
+package com.epam.reportportal.service.analytics.item;
+
+import com.google.common.collect.Maps;
+import io.reactivex.annotations.Nullable;
+
+import java.util.Map;
+
+/**
```
[the remark this answers is not in the document]
Comment (Yumfriez) [reply src/main/java/com/epam/reportportal/service/analytics/item/AnalyticsEvent.java:24]: added

File: src/main/java/com/epam/reportportal/service/analytics/GoogleAnalytics.java [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+
+	public GoogleAnalytics(Scheduler scheduler, String trackingId) {
+		this.scheduler = scheduler;
+		this.baseUrl = DEFAULT_BASE_URL;
+		Collections.addAll(defaultRequestParams,
+				new BasicNameValuePair("de", "UTF-8"),
+				new BasicNameValuePair("v", "1"),
+				new BasicNameValuePair("cid", UUID.randomUUID().toString()),
+				new BasicNameValuePair("tid", trackingId)
+		);
```
[the remark this answers is not in the document]
Comment (Yumfriez) [reply src/main/java/com/epam/reportportal/service/analytics/GoogleAnalytics.java:74]: fixed

File: src/main/java/com/epam/reportportal/service/analytics/GoogleAnalytics.java [hunk 8]
```diff
@@ -0,0 +1,132 @@
...
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.Closeable;
+import java.nio.charset.StandardCharsets;
+import java.util.*;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+/**
```
[the remark this answers is not in the document]
Comment (Yumfriez) [reply src/main/java/com/epam/reportportal/service/analytics/GoogleAnalytics.java:44]: added

**Pull request merged**